usr/lib/ica_s390_stdll/ica_specific.c

```diff
@@ -2413,132 +2509,116 @@ static CK_RV ica_specific_rsa_decrypt(STDLL_TokData_t *tokdata,
+        /* mech_rsa.c:ckm_rsa_decrypt accepts only CKO_PRIVATE_KEY,
+         * but Private Key can have 2 representations (see PKCS#1):
+         *  - Modulus + private exponent
+         *  - p, q, dp, dq and qInv (CRT format)
+         * The former should use ica_rsa_key_mod_expo_t and the latter
+         * ica_rsa_key_crt_t. Detect what representation this
+         * key_obj has and use the proper convert function */
+
+        template_attribute_get_non_empty(key_obj->template, CKA_MODULUS,
+                                         &modulus);
+        template_attribute_get_non_empty(key_obj->template,
+                                         CKA_PRIVATE_EXPONENT, &priv_exp);
+        template_attribute_get_non_empty(key_obj->template, CKA_PRIME_1,
+                                         &prime1);
+        template_attribute_get_non_empty(key_obj->template, CKA_PRIME_2,
+                                         &prime2);
+        template_attribute_get_non_empty(key_obj->template, CKA_EXPONENT_1,
+                                         &exp1);
+        template_attribute_get_non_empty(key_obj->template, CKA_EXPONENT_2,
+                                         &exp2);
+        template_attribute_get_non_empty(key_obj->template, CKA_COEFFICIENT,
+                                         &coeff);
+
+        /* Need to check for CRT Key format *BEFORE* check for mod_expo key,
+         * that's because opencryptoki *HAS* a CKA_PRIVATE_EXPONENT attribute
+         * even in CRT keys (but with zero length) */
+
+        if (modulus && prime1 && prime2 && exp1 && exp2 && coeff) {
+            /* ica_rsa_key_crt_t representation */
+            ex_data->crtKey = rsa_convert_crt_key(modulus, prime1, prime2,
+                                                  exp1, exp2, coeff);
+            if (ex_data->crtKey == NULL) {
+                TRACE_ERROR("rsa_convert_crt_key failed\n");
+                rc = CKR_FUNCTION_FAILED;
+                goto done;
+            }
+            /* same check as above */
+            if (ex_data->crtKey->key_length != in_data_len) {
+                TRACE_ERROR("%s\n", ock_err(ERR_DATA_LEN_RANGE));
+                rc = CKR_ENCRYPTED_DATA_LEN_RANGE;
+                goto done;
+            }
+        } else if (modulus && priv_exp) {
+            /* ica_rsa_key_mod_expo_t representation */
+            ex_data->modexpoKey = rsa_convert_mod_expo_key(modulus, NULL,
+                                                           priv_exp);
+            if (ex_data->modexpoKey == NULL) {
+                TRACE_ERROR("rsa_convert_mod_expo_key failed\n");
+                rc = CKR_FUNCTION_FAILED;
+                goto done;
+            }
+            /* in_data must be in big endian format. Size in bits must not
+             * exceed the bit length of the key, and size in bytes must
+             * be the same */
+            // FIXME: we're not cheking the size in bits of in_data
```

Comment:
Do we need to remove the FIXME comment? Or is it intended to be left in the code?

Reply:
Yea, I just moved that comment with the other lines into the block.... 
But, yes, lets remove the FIXME here. 
There are other similar FIXMEs in the code, I leave them as is since I am not touching those lines.